lib/date_time/date_time_core.c

```diff
@@ -53,39 +56,46 @@ static void date_time_core_notify_event(enum date_time_evt_type time_source)
 	}
 }
 
-static void date_time_core_schedule_update(bool check_pending)
+static void date_time_core_schedule_update(bool fast)
 {
-	/* (Re-)schedule time update work
-	 * if periodic updates are requested in the configuration.
-	 */
-	if (CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS > 0) {
-		/* Don't reschedule time update work in some cases,
-		 * e.g. if time is not obtained and the work is already pending.
-		 */
-		if (check_pending && k_work_delayable_is_pending(&date_time_update_work)) {
-			return;
-		}
+	int delay = CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS;
+
+	/* If we have already exhausted the maximum fast retries, switch to non-fast. */
```

Comment:
```suggestion
	/* If we have already exhausted the maximum fast retries, switch to normal. */
```
would prefer normal over non-fast